src/libopensc/card-nqApplet.c

```diff
@@ -0,0 +1,556 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "asn1.h"
+#include "cardctl.h"
+#include "internal.h"
+#include "log.h"
+
+#define APPLET_VERSION_LEN 2
+#define APPLET_MEMTYPE_LEN 1
+#define APPLET_SERIALNR_LEN 8
+
+/* card constants */
+static const struct sc_atr_table nqapplet_atrs[] = 
+{
+	{ "3b:d5:18:ff:81:91:fe:1f:c3:80:73:c8:21:10:0a", NULL, NULL, SC_CARD_TYPE_NQ_APPLET, 0, NULL },
+	{ NULL, NULL, NULL, 0, 0, NULL }
+};
+
+static const u8 nqapplet_aid[] = {0xd2, 0x76, 0x00, 0x01, 0x80, 0xBA, 0x01, 0x44, 0x02, 0x01, 0x00};
+
+static struct sc_card_operations nqapplet_operations;
+static struct sc_card_operations *iso_operations = NULL;
+
+#define KEY_REFERENCE_NO_KEY	0x00
+#define KEY_REFERENCE_AUTH_KEY	0x01
+#define KEY_REFERENCE_ENCR_KEY	0x02
+
+struct nqapplet_driver_data
+{
+	u8 version_minor;
+	u8 version_major;
+	u8 key_reference;
+	u8 serial_nr[APPLET_SERIALNR_LEN];
+};
+typedef struct nqapplet_driver_data * nqapplet_driver_data_ptr;
+
+static struct sc_card_driver nqapplet_driver = 
+{
+	"NQ-Applet",      // name
+	"nqapplet",      // short name
+	&nqapplet_operations,	// operations
+	NULL,           // atr table
+	0,              // nr of atr
+	NULL            // dll?
+};
+
+static const struct sc_card_error nqapplet_errors[] = 
+{
+	{ 0x6700, SC_ERROR_WRONG_LENGTH, "Invalid LC or LE"}, 
+	{ 0x6982, SC_ERROR_SECURITY_STATUS_NOT_SATISFIED, "Security status not satisfied" }, //TODO MK/DK??
+	{ 0x6985, SC_ERROR_NOT_ALLOWED,	"Invalid PIN or key" },
+	{ 0x6986, SC_ERROR_NOT_ALLOWED,	"Conditions of use not satisfied" },
+	{ 0x6A80, SC_ERROR_INVALID_ARGUMENTS, "Invalid parameters" },
+	{ 0x6A82, SC_ERROR_OBJECT_NOT_FOUND, "Data object not found"},
+	{ 0x6A84, SC_ERROR_NOT_ENOUGH_MEMORY, "Not enough memory" },
+	{ 0x6A86, SC_ERROR_INCORRECT_PARAMETERS, "Invalid P1 or P2" }, 
+	{ 0x6A88, SC_ERROR_INVALID_ARGUMENTS, "Wrong key ID" },
+	{ 0x6D00, SC_ERROR_FILE_NOT_FOUND, "Applet not found" }
+};
+
+/* convenience functions */
+
+static int init_driver_data(sc_card_t *card, u8 version_major, u8 version_minor, u8 * serial_nr, size_t cb_serial_nr)
+{
+	nqapplet_driver_data_ptr data = calloc(1, sizeof(struct nqapplet_driver_data));
+	if (data == NULL)
+	{
+		LOG_FUNC_RETURN(card->ctx, SC_ERROR_OUT_OF_MEMORY);
+	}
```

Comment:
Please, use the braces formatting as in the rest of the code. Some formal hints/references might be found in #2017, even though they are not finalized yet.

Reply:
Adjusted braces formatting as in rest of the code.

Reply:
I do not see this addressed yet. You can run the `clang-format` with the configuration attached to the #2017 to get close to the expected formatting.

Reply:
I had another clang-format configuration, updated it and reformatted the files.